src/background.ts

```diff
@@ -6,15 +6,34 @@
     const calendar_name = "AutoCalendar Demo";
     const autocalendar = new AutoCalendar();
 
-    chrome.runtime.onMessage.addListener(
-        async (request, sender, sendResponse) => {
-            if (request.action === "sync_canvas") {
+    let prev_execution = null;
+    let isRunning = false;
+
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+        if (request.action === "oauth") {
+            autocalendar.calendar.session.oauth_token();
+        } else if (request.action === "sync_canvas") {
+            let time_diff =
+                prev_execution === null
+                    ? Number.MAX_VALUE
+                    : new Date().getTime() - prev_execution.getTime();
+            if (!isRunning && time_diff > 62000) {
+                isRunning = true;
                 console.time("sync_canvas");
-                await autocalendar.sync_canvas(calendar_name);
-                console.timeEnd("sync_canvas");
-            } else if (request.action == "oauth") {
-                await autocalendar.calendar.session.oauth_token();
+                autocalendar.sync_canvas(calendar_name).then(() => {
+                    console.timeEnd("sync_canvas");
```

Comment:
console.timeEnd is to time the function, it doesn't make sense to interleave it with the rest of the code. We should probably move it to the end of the function after sendResponse().

Reply:
Sounds good.